### ISSUE TITLE
Unable to compile model for NNPA

### ISSUE THREAD

**Issue by cjvolzka** [onnx/onnx-mlir#1345]
After pulling in latest onnx-mlir commit (11e0971f4037bab7f3e7efe3608f2dcdc3426dff) we can no longer compile the [ccf_lstm_dyn_tf2onnx_new.onnx](https://github.ibm.com/ai-on-z/in-house-onnx-models/blob/main/TF-ONNX/ccf_lstm_dyn_tf2onnx_new.onnx) model when `--maccel=NNPA` is set. Previously this model compiled fine when using 9f04c01d626155d9e6ad23f3b37e6a4568956056.

Note: I get the same error when using our build of onnx-mlir (which sets `-DACCELERATORS_TO_BUILD="NNPA"` when building onnx-mlir and using the image https://hub.docker.com/r/onnxmlirczar/onnx-mlir image which was updated about an hour ago.

```
onnx-mlir --EmitLib --O3 --mcpu=z14 --maccel=NNPA --mtriple=s390x-ibm-loz ccf_lstm_dyn_tf2onnx_new.onnx -v
onnx-mlir: /workdir/llvm-project/mlir/lib/Dialect/Affine/Utils/Utils.cpp:1108: mlir::LogicalResult mlir::replaceAllMemRefUsesWith(mlir::Value, mlir::Value, mlir::Operation*, llvm::ArrayRef<mlir::Value>, mlir::AffineMap, llvm::ArrayRef<mlir::Value>, llvm::ArrayRef<mlir::Value>, bool): Assertion `false && "multiple dereferencing uses in a single op not supported"' failed.
Aborted (core dumped)
```

**Comment by cjvolzka** [comment]
Follow up: I've gotten up to 35f9742c62c78477b003881c3f58d2dde5a295b3 and not have hit the issue. So it must have been introduced at or after 2ebd9eb4f95250284d90a664830f8ede64894ee3 and exists in 11e0971f4037bab7f3e7efe3608f2dcdc3426dff

**Comment by cjvolzka** [comment]
Looks like issue first appears in 2ebd9eb4f95250284d90a664830f8ede64894ee3

**Comment by AlexandreEichenberger** [comment]
@tungld do you mind looking into it, it appears to have been introduced by CSE PR.

**Comment by AlexandreEichenberger** [comment]
@tungld the CSE is useful, so I assume that you will look into what the problem is, in order to reintroduce it.

**Comment by tungld** [comment]
@AlexandreEichenberger Yes, I will investigate more about the issue.

**Comment by tungld** [comment]
Reopen this to investigate its root cause.

**Comment by cjvolzka** [comment]
> Reopen this to investigate its root cause.

@tungld is this Issue still under investigation. https://github.com/onnx/onnx-mlir/pull/1361 has merged but I'm unsure if there was more you were looking into.

**Comment by tungld** [comment]
@cjvolzka it was completely fixed by #1361. I forgot to close this. Closing it now. Thanks!
